src/ir_Corona.cpp

```diff
@@ -191,45 +188,46 @@ bool IRCoronaAc::validSection(const uint8_t state[], const uint16_t pos,
   if ((section % kCoronaAcSections) * kCoronaAcSectionBytes != pos)
     return false;
   // all individual sections has the same prefix
-  if (state[pos + kCoronaAcSectionHeader0Pos] != kCoronaAcSectionHeader0) {
+  auto p = reinterpret_cast<const CoronaSection*>(state + pos);
```

Comment:
I'm not familiar enough with how compilers treat `auto` between implementations and platforms/environments.
i.e. I don't trust it enough that it isn't different on the ESP platform and the linux/X86 platform etc that the unit tests run on.
Is it possible to avoid it's use and stick to strictly to `uintN_t` sized variables?